draft-wussler-openpgp-pqc.md

```diff
@@ -706,29 +741,57 @@ The placeholder `eccKem` has to be replaced with the specific ECC-KEM from the
 row "ECC-KEM" of {{tab-ecdh-cfrg-artifacts}}, {{tab-ecdh-nist-artifacts}}, and
 {{tab-ecdh-brainpool-artifacts}}.
 
-#### X25519-KEM and X448-KEM {#x25519-x448-kem}
+#### X25519-KEM {#x25519-kem}
+
+The encapsulation and decapsulation operations of `x25519kem` are described
+using the function `X25519()` and encodings defined in [RFC7748]. The
+`eccPrivateKey` is denoted as `r`, the `eccPublicKey` as `R`, they are subject
+to the equation `R = X25519(r, 9)`.
 
-The operations `x25519Kem.encap()` and `x448Kem.encap()` are defined as
-follows:
+The operation `x25519Kem.encap()` is defined as follows:
 
- 1. Generate an ephemeral key pair {`v`, `V=vG`} according to
-    the procedure defined in [RFC7748], that is, without any
-    measures such as masking or clamping
+ 1. Generate an ephemeral key pair {`v`, `V`} via `V = X25519(v,9)`
 
- 2. Compute the shared point `X = vR` where `R` is the component public key
-    `eccPublicKey` according to [RFC7748]
+ 2. Compute the shared coordinate `X = X25519(v, R)` where `R` is the public key
+    `eccPublicKey`
 
  3. Set the output `eccCipherText` to `V`
 
- 4. Set the output `eccKeyShare` to `X` in native format
+ 4. Set the output `eccKeyShare` to `SHA3-256(X || eccCipherText)`
 
-The operations `x25519Kem.decap()` and `x448Kem.decap()` are defined as
-follows:
+The operation `x25519Kem.decap()` is defined as follows:
 
- 1. Compute the shared Point `X = rV` where `r` is the `eccPrivateKey` and `V`
-    is the `eccCipherText` according to [RFC7748]
+ 1. Compute the shared coordinate `X = X25519(r, V)`, where `r` is the
+    `eccPrivateKey` and `V` is the `eccCipherText`
 
- 2. Set the output `eccKeyShare` to `X` in native format
+ 2. Set the output `eccKeyShare` to `SHA3-256(X || eccCipherText)`
+
+#### X448-KEM {#x448-kem}
+
+The encapsulation and decapsulation operations of `x448kem` are described using
+the function `X448()` and encodings defined in [RFC7748]. The `eccPrivateKey`
+is denoted as `r`, the `eccPublicKey` as `R`, they are subject to the equation
+`R = X448(r, 5)`.
+
+The operation `x448.encap()` is defined as follows:
+
+ 1. Generate an ephemeral key pair {`v`, `V`} via `V = X448(v,5)`
+
+ 2. Compute the shared coordinate `X = X448(v, R)` where `R` is the public key
+    `eccPublicKey`
+
+ 3. Set the output `eccCipherText` to `V`
+
+ 4. Set the output `eccKeyShare` to the first 56 octets of `SHA3-512(X ||
+    eccCipherText)`
```

Comment:
I'm personally not the biggest fan of this change. Given that the length of the ciphertexts is guaranteed by protocol construction I would not add a hashing here, that's expensive!

Also, I find silly truncating the hash, when anyway it'll end up in a KDF.

Reply:
I agree that there is potentially an optimization possible. However, we have seen that the original draft was actually insecure (the ECC part was not CCA2 secure). I would adopt the change. This definitely gives you security in a modular way.  In my opinion we can optimize it if we manage to directly prove that the optimization is also secure.

Reply:
@ahuelsing don't get me wrong, I am happy with `X ||  eccCipherText`, but I would remove the `SHA-3-512` here (and in the other algorithms too)

Reply:
And then you do not have a secure KEM.